pkg/storage/snapshot/snapshot.go

```diff
@@ -759,6 +760,8 @@ func (ctrl *VMSnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Vi
 		}
 	}
 
+	metrics.NewVMSnapshotCreated(vmSnapshotCpy)
```

Comment:
Why don't you want to chime in the metrics handling in the code block that handles successful VM snapshot in the `updateSnapshotStatus` function? for example [here](https://github.com/kubevirt/kubevirt/blob/64bc38790f9bd3819302d621674cbf16df92f5f4/pkg/storage/snapshot/snapshot.go#L705)